app/controllers/ips_controller.rb

```diff
@@ -0,0 +1,80 @@
+class IpsController < ApplicationController
+  before_action :check_format, :check_params, :set_ip, only: [:create]
+
+  # GET /compute
+  def index
+    if Ip.count > 0
+      raw_ips = Ip.all
+      ips = serialize(raw_ips)
+
+      Ip.destroy_all
+      ComputeHistory.create(result: ips.to_json)
+
+      render json: ips
+    else
+      render nothing: true, status: :no_content
+    end
+
+  end
+
+  # POST /store
+  def create
+    @ip ||= Ip.new(ip_params)
+
+    if @ip.save
+      numbers = params[:numbers]
+
+      numbers.each do |value|
+        number = Number.new(value: value)
+        @ip.numbers << number unless @ip.numbers.find_by value: value
+      end
+
+      render json: @ip, status: :created
+    else
+      render json: @ip.errors, status: :unprocessable_entity
+    end
+  end
+
+  private
+
+    def set_ip
+      @ip = Ip.find_by address: params[:address]
+    end
+
+    def ip_params
+      params.require(:ip).permit(:address)
+    end
+
+    def check_format
+      render nothing: true, status: 415 unless request.headers["Content-Type"] =~ /json/
+    end
+
+    def check_params
+      if params[:address].blank? || params[:numbers].blank?
+        render nothing: true, status: :bad_request
+      elsif check_address(params[:address])
+        render nothing: true, status: :bad_request
+      end
+    end
+
+    def check_address(address)
+      address.split('.').each do |portion|
+        if portion.to_i < 0 || portion.to_i > 255 || portion.scan(/\D/).any?
+          return true
+        end
+      end
+
+      return false
+    end
+
+    def serialize(raw_ips)
+      ips = []
+
+      raw_ips.each do |ip|
+        ips << { address: ip.address, numbers: ip.numbers_value }
+      end
+
+      ips
+    end
+
+end
```

Comment:
Your controller is handling too many responsibilities. Especially at the private methods. 
A simple tip is that it could be divided between modules like helpers, serializers, validators and so on.

Also, it's following the "rails web app way" to create an API. When you need to include a comment to say that `def index` will respond to `GET /compute` there is a bad smell here. Your code and your test suite should be the documentation of what is happening by itself, without needing extra description.

Reply:
Agreed,
I thought the size of code was too small to do so. But I agree that it is a good practice to modularize the code. 

About the comment on the action declaration, I agree and that code should be refactored.